styx-api-service/src/main/java/com/spotify/styx/api/StatusResource.java

```diff
@@ -144,27 +154,28 @@ private static String arg(String name, RequestContext rc) {
     return Response.forPayload(RunStateDataPayload.create(runStates));
   }
 
-  private Map<WorkflowInstance, RunState> getActiveStates(String componentsStr) throws IOException {
-    final List<String> components = Arrays.asList(componentsStr.split(","));
-    final ImmutableMap.Builder<WorkflowInstance, RunState> mapBuilder = ImmutableMap.builder();
-
-    final List<Optional<IOException>> exceptions = components.parallelStream().map( componentId -> {
-      Optional<IOException> exception = Optional.empty();
-      try{
-        final Map<WorkflowInstance, RunState> stateMap = storage.readActiveStates(componentId);
-        for (WorkflowInstance instance : stateMap.keySet()) {
-          mapBuilder.put(instance, stateMap.get(instance));
-        }
-      } catch (IOException e) {
-        exception = Optional.of(e);
-      }
-      return exception;
-      }).filter(Optional::isPresent).collect(toList());
-    if (!exceptions.isEmpty()) {
-      throw exceptions.get(0).orElseThrow();
-    }
+  private Map<WorkflowInstance, RunState> getActiveStates(String componentsStr) {
+    var components = Arrays.asList(componentsStr.split(","));
+    var activeStatesOrExceptions =
```

Comment:
Not problem of this PR, but we recently did a change to simplify FJP usage: https://github.com/spotify/styx/pull/795/files#diff-0cec30db46b18b2c0132f746e3bad5b0R453

We can do the same here in this PR.